src/Compilers/Test/Utilities/CSharp/CSharpTestBase.cs

```diff
@@ -2411,6 +2411,22 @@ protected static CSharpCompilation CreateCompilationWithSpanAndMemoryExtensions(
                     parseOptions: parseOptions);
             }
         }
+
+        internal static string GetIdForErrorCode(ErrorCode code)
+        {
+            return MessageProvider.Instance.GetIdForErrorCode((int)code);
+        }
+
+        internal static ImmutableDictionary<string, ReportDiagnostic> ReportStructInitializationWarnings { get; } = ImmutableDictionary.CreateRange(
```

Comment:
>ReportStructInitializationWarnings [](http://example.com/codeflow?start=70&length=34)

Let's use warning waves (`WithWarningLevel(int)`) to control the diagnostics in tests. We shouldn't need this list of diagnostics.

Reply:
I don't think this is right. We actually don't want these diagnostics to be reported even if the user is on the latest wave. They need to opt-in to something more "severe". Tracked in #60176